### PR TITLE
Implement Osmo Liquid Pooler Tests

### LINKED ISSUE #237: osmo liquid pooler local interchaintest
blocked by #230 

currently our osmo liquid pooler testing is done with interchaintest as part of the two party POL test.

with local-interchain in place, we can do better.

this should involve, but not be limited to, testing various edge cases related to relayers.

we should test timeouts, relayers going down, and anything related to polytone.

### PR DESCRIPTION
**Pull request by lexzaiello**
Closes #237 . This PR implements tests for polytone proxy creation in the osmo liquid pooler. This test ensures that the liquid pooler handles a killed relayer and can recover after failing to create the proxy.

### CODE REVIEW

File: local-interchaintest/src/helpers/constants.rs [hunk 1]
```diff
@@ -10,6 +11,9 @@ pub const POLYTONE_PATH: &str = "local-interchaintest/contracts/polytone";
 pub const EXECUTE_FLAGS: &str = "--gas=auto --gas-adjustment=3.0";
 pub const OSMOSIS_FEES: &str = "--fees=100000uosmo";
 
+pub const ACC0_ADDRESS_GAIA: &str = "cosmos1hj5fveer5cjtn4wd6wstzugjfdxzl0xpxvjjvr";
+pub const ACC0_ADDRESS_NEUTRON: &str = "neutron1hj5fveer5cjtn4wd6wstzugjfdxzl0xpznmsky";
+pub const ACC0_ADDRESS_OSMO: &str = "osmo1hj5fveer5cjtn4wd6wstzugjfdxzl0xpwhpz63";
```
Comment (keyleu) [inline local-interchaintest/src/helpers/constants.rs:16]: nit: I never added these ACC0 because we already have them in localic-utils as the CHAIN_ADMIN_ADDR, you can reuse those if you want or leave it like this

File: local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    // Combined test: runs test_timeout and test_ok in sequence with the same pooler
+    // to ensure that the pooler can recover from a timeout
+    test_pooler_timeout_recover(test_ctx, &poolers.2)?;
+
+    info!("Finished osmo liquid pooler tests!");
+
+    Ok(())
+}
+
+fn start_relayer(test_ctx: &mut TestContext) {
```
Comment (keyleu) [inline local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs:81]: we can remove these 2 functions and use the new ones you added to localic-utils right?

File: local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+
+/// Tests that the pooler does not advance when there is no relayer available.
+/// Leaves the relayer intact after its execution.
+fn test_pooler_timeout(test_ctx: &mut TestContext, pooler: &str) -> Result<(), LocalError> {
+    // Stop the relayer
+    stop_relayer(test_ctx);
+
+    let neutron = test_ctx.get_chain(NEUTRON_CHAIN_NAME);
+
+    let pooler_contract =
```
Comment (keyleu) [inline local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs:353]: since you already have the contract address (you are passing it as an argument). You can reuse my helper function I created in common.rs to query the contract state, up to you, but you can remove a lot of code here that way. I also made one for ticking so you can remove pretty much all the tick/query code here and use those.
I forgot to mention this in other PRs because you also query state / tick there.
This will also remove a lot of duplication everywhere.

File: local-interchaintest/src/helpers/constants.rs (hunk 1 above)
Comment (dowlandaiello) [reply local-interchaintest/src/helpers/constants.rs]: I will reuse the ones in localic-utils! Just pushed the change.

File: local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs (hunk 2 above)
Comment (dowlandaiello) [reply local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs]: Pushed the change!

File: local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs (hunk 3 above)
Comment (dowlandaiello) [reply local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs]: Pushed the change! Going to fix this redundancy in the other tests too in a PR after this one.

Comment (dowlandaiello) [reply local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs]: Just ran the tests locally btw on the latest commit. Everything seems to be working!

Comment (keyleu) [reply local-interchaintest/src/tests/osmo_liquid_pooler/osmo_liquid_pooler.rs]: nice!

**Review by keyleu — APPROVED** [review]
